File: data-munging/src/main/java/pro/verron/datamunging/Parsers.java
package pro.verron.datamunging;

public class Parsers {
    public static final Parser<Integer> INTEGER_PARSER = of(Integer::parseInt);
    public static final Parser<Double> DOUBLE_PARSER = of(Double::parseDouble);
    public static final Parser<String> STRING_PARSER = of(String::trim);
    public static <U> Parser<U> of(Parser<U> parser) {
        return line -> line.trim().transform(parser::parse);
    }
}

File: data-munging/src/main/java/pro/verron/datamunging/Parsers.kt
package pro.verron.datamunging

import java.util.function.Function

object Parsers {
    val INTEGER_PARSER = of { s: String -> s.toInt() }
    val DOUBLE_PARSER = of { s: String -> s.toDouble() }
    val STRING_PARSER = of { s: String -> s }

    private fun <U> of(parser: Function<String, U>): Function<String, U> {
        return Function { line: String -> line.trim().transform(parser) }
    }
}